google/cloud/storage/examples/storage_bucket_samples.cc

```diff
@@ -548,6 +548,156 @@ void GetServiceAccountForProject(google::cloud::storage::Client client,
   //! [get service account for project]
   (std::move(client), project_id);
 }
+
+void GetDefaultEventBasedHold(google::cloud::storage::Client client, int& argc,
+                              char* argv[]) {
+  if (argc != 2) {
+    throw Usage{"get-default-event-based-hold <bucket-name>"};
+  }
+  auto bucket_name = ConsumeArg(argc, argv);
+  //! [get default event based hold]
+  // [START storage_get_default_event_based_hold]
+  namespace gcs = google::cloud::storage;
+  [](gcs::Client client, std::string bucket_name) {
+    gcs::BucketMetadata metadata = client.GetBucketMetadata(bucket_name);
+    std::cout << "The default event-based hold for objects in bucket "
+              << bucket_name << " is "
+              << (metadata.default_event_based_hold() ? "enabled" : "disabled")
```

Comment:
Issue with this: At the moment, If a user performs a `client.GetBucketMetadata(bucket_name, Fields("name"))` `default_event_based_hold()` will return `false` when the value is unknown because it's set by default as `false` and it wasn't requested from the service. Give the way a developer would interact with this library I don't think it will be a large concern. However, I do suspect someone may hit it by accident though.


_Side note_: I just [these](https://github.com/GoogleCloudPlatform/google-cloud-cpp/blob/5252992d27529c82d76135027e368e22de834376/google/cloud/storage/internal/generic_request.h#L179) well known parameters aren't documented along with other well known parameters.